vcd/datasource_vcd_independent_disk_test.go

```diff
@@ -0,0 +1,168 @@
+// +build disk ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/terraform"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+// Test independent disk data resource
+// Using a disk data source we reference a disk data source
+func TestAccVcdDataSourceIndependentDisk(t *testing.T) {
+	resourceName := "TestAccVcdDataSourceIndependentDisk_1"
+	datasourceName := "TestAccVcdDataSourceIndependentDisk_Data"
+	datasourceNameWithId := "TestAccVcdDataSourceIndependentDiskWithId_Data"
+	diskName := "TestAccVcdDataSourceIndependentDisk"
+
+	var params = StringMap{
+		"Vdc":                  testConfig.VCD.Vdc,
+		"name":                 diskName,
+		"description":          diskName + "description",
+		"size":                 "5242880",
+		"busType":              "SCSI",
+		"busSubType":           "lsilogicsas",
+		"storageProfileName":   "*",
+		"ResourceName":         resourceName,
+		"Tags":                 "disk",
+		"dataSourceName":       datasourceName,
+		"datasourceNameWithId": datasourceNameWithId,
+	}
+
+	configText := templateFill(testAccCheckVcdDataSourceIndependentDisk, params)
+	params["FuncName"] = t.Name() + "-withId"
+	configTextWithId := templateFill(testAccCheckVcdDataSourceIndependentDiskWithId, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testDiskResourcesDestroyed,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:             configText,
+				ExpectNonEmptyPlan: true,
```

Comment:
Do we really need it?
I think the data source config could have a dependency in such way:
```hcl
data "vcd_independent_disk" "{{.dataSourceName}}" {
  name    = "${vcd_independent_disk.{{.ResourceName}}.name}"
}
```

Then it should not have a non empty plan after run.

https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies

Reply:
tx, updated